src/main/java/htsjdk/variant/vcf/VCFUtils.java

```diff
@@ -157,21 +159,77 @@ public static VCFHeader withUpdatedContigs(final VCFHeader oldHeader, final File
      */
     public static List<VCFContigHeaderLine> makeContigHeaderLines(final SAMSequenceDictionary refDict,
                                                                   final File referenceFile) {
-        final List<VCFContigHeaderLine> lines = new ArrayList<VCFContigHeaderLine>();
+        final List<VCFContigHeaderLine> lines = new ArrayList<>();
         final String assembly = referenceFile != null ? getReferenceAssembly(referenceFile.getName()) : null;
         for ( final SAMSequenceRecord contig : refDict.getSequences() )
             lines.add(makeContigHeaderLine(contig, assembly));
         return lines;
     }
 
     private static VCFContigHeaderLine makeContigHeaderLine(final SAMSequenceRecord contig, final String assembly) {
-        final Map<String, String> map = new LinkedHashMap<String, String>(3);
+        final Map<String, String> map = new LinkedHashMap<>(3);
         map.put("ID", contig.getSequenceName());
         map.put("length", String.valueOf(contig.getSequenceLength()));
         if ( assembly != null ) map.put("assembly", assembly);
         return new VCFContigHeaderLine(map, contig.getSequenceIndex());
     }
 
+    /**
+     * This method creates a temporary VCF file and it's appropriately named index file, and will delete them on exit.
+     * @param prefix - The prefix string to be used in generating the file's name; must be at least three characters long
+     * @param suffix - The suffix string to be used in generating the file's name; may be null, in which case the suffix ".tmp" will be used
+     * @return A File object referencing the newly created temporary VCF file
+     * @throws IOException - if a file could not be created.
+     */
+    public static File createTemporaryIndexedVcfFile(final String prefix, final String suffix) throws IOException {
```

Comment:
I'll opt to return a `java.nio.Path` to continue moving forward to `Path` representation of files insted of `File`...

Reply:
enough parts still do not support path...so this is still non-trivial...I'd prefer not to mix up the different purposes.

Reply:
If the code moves to Path for parts using this method, the File version should be replicated introducing unexpected problems (as unfortunately I did with the sorting collections). Thus, it's simpler to implement directly methods with Path and use the Path.toFile/File.toPath methods if required... to be sure that it's less painful to evolve to a pure Path htsjdk (something that I'm currently  kind of working on)...